### PR TITLE
This PR resolves Issue 217

### PR DESCRIPTION
**Pull request by ghost**
This PR relates to #217

The pull request consists of:

* Changes in AbstractIntegrationTest,  RnaExpressionFastqIntegrationTest and thymeleaf templates to run RnaExpressionFastqIntegrationTest in master mode
* New thymeleaf template master_template_test for validation master script

### CODE REVIEW

File: RNAexpression_Fastq_test/sh_files/RnaExpression_Fastq_alignment_for_SW1271_analysis.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#!/bin/bash -x
```
Comment (kamyshova) [inline RNAexpression_Fastq_test/sh_files/RnaExpression_Fastq_alignment_for_SW1271_analysis.sh]: As I can see, this file and below (all in RNAexpression_Fastq_test
folder) are your test files and should be deleted from PR.

File: example/study_config/config_RnaExpression_Fastq_test.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [Series_Info]
 job_name = pe_job
-dir_out = /ngs/data/demo/test/RNAexpression_Fastq_test
-fastq_list = /ngs/data/demo/test/example/RnaExpression_RNASeq_SampleFastqPaths.txt
+dir_out = RNAexpression_Fastq_test
+fastq_list = C:\Users\Aleksei_Zaichenkov\IdeaProjects\fonda\example\sample_manifest\RnaExpression_RNASeq_SampleFastqPaths.txt
```
Comment (kamyshova) [inline example/study_config/config_RnaExpression_Fastq_test.txt]: This is an example study file. Please, clean up your changes from PR.

File: src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java [hunk 3]
```diff
@@ -90,20 +92,38 @@ public void cleanOutputDirForNextTest(String outputDir) throws IOException {
...
             writeToFile(globalConfig, source, lineSeparator);
+
+            if (ArrayUtils.isNotEmpty(nonRequiredOptions) && Arrays.asList(nonRequiredOptions).contains("-master")) {
+                context.setVariable("master", true);
+            } else {
+                context.setVariable("master", false);
+            }
+
+            final String[] args = ArrayUtils.addAll(
+                    new String[]{"-test", "-global_config", globalConfig, "-study_config", studyConfig},
```
Comment (kamyshova) [inline src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java]: There is no need to create `new String[]{}`.

File: src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java [hunk 4]
```diff
@@ -15,101 +15,138 @@
...
+    private static final String SAMPLE_NAME = "smv1/";
+    private static final Integer NUMBER_OF_SCRIPTS = 6;
+
+    private static final Set<String> CONFIG_WITH_POST_PROCESS = new HashSet<>(Arrays.asList(
+            RNA_EXPRESSION_FASTQ_G_STAR_WITH_RSEM,
+            RNA_EXPRESSION_FASTQ_G_STAR_WITHOUT_RSEM,
+            RNA_EXPRESSION_FASTQ_G_RSEM_WITHOUT_HISAT2)
+    );
+    private static final String[] EXPECTED_SCRIPTS = new String[]{
+        "",
```
Comment (kamyshova) [inline src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java]: Do we need an empty string here?

File: src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Hisat2.txt [hunk 5]
```diff
@@ -84,7 +84,7 @@ else
 	echo `date` The job was aborted due to ERRORS found.
 	exit 1;
 fi
-
+[# th:switch = "${master}"][# th:case = "false"]
```
Comment (kamyshova) [inline src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Hisat2.txt]: Why don't we simplify the check to [# th:if = "${!master}"]?

File: src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Salmon.txt [hunk 6]
```diff
@@ -51,7 +51,7 @@ else
 	echo `date` The job was aborted due to ERRORS found.
 	exit 1;
 fi
-
+[# th:switch = "${master}"][# th:case = "false"]
```
Comment (kamyshova) [inline src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Salmon.txt]: The same as above and below

File: src/test/resources/templates/library_csv_template_output.txt [hunk 7]
```diff
@@ -1,4 +1,4 @@
 fastqs,sample,library_type
-/path/to/sampleName,sampleName,Gene Expression
-/path/to/sampleName,sampleName,Antibody Capture
+[(${path})]sampleName,sampleName,Gene Expression
+[(${path})]sampleName,sampleName,Antibody Capture
```
Comment (kamyshova) [inline src/test/resources/templates/library_csv_template_output.txt]: These template changes do not match test changes. Test falls

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java [hunk 8]
```diff
@@ -120,124 +157,232 @@ void testRnaExpressionFastq(final String gConfigPath, final String outputShFile,
         assertEquals(expectedCmd.trim(), getCmd(outputShFile).trim());
     }
 
-    @SuppressWarnings("PMD")
+    @ParameterizedTest(name = "{3}-test")
+    @MethodSource("initParameters")
+    void testRnaExpressionFastqMaster(final String gConfigPath, final String outputShFile,
+                                      final String templatePath, final String expectedBaseScript,
+                                      final String expectedSecondScript) throws IOException, URISyntaxException {
+        startAppWithConfigs(gConfigPath, S_CONFIG_PATH, new String[]{"-master"});
```
Comment (kamyshova) [inline src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java]: Add spaces to "new String[] { "-master" }", please

File: RNAexpression_Fastq_test/sh_files/RnaExpression_Fastq_alignment_for_SW1271_analysis.sh (hunk 1 above)
Comment (ghost) [reply RNAexpression_Fastq_test/sh_files/RnaExpression_Fastq_alignment_for_SW1271_analysis.sh]: Fixed

File: example/study_config/config_RnaExpression_Fastq_test.txt (hunk 2 above)
Comment (ghost) [reply example/study_config/config_RnaExpression_Fastq_test.txt]: Done

File: src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java (hunk 4 above)
Comment (ghost) [reply src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java]: Removed

File: src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Hisat2.txt (hunk 5 above)
Comment (ghost) [reply src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Hisat2.txt]: Done

File: src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Salmon.txt (hunk 6 above)
Comment (ghost) [reply src/integrationTest/resources/templates/RnaExpressionFastq/rnaExpression_Fastq_Salmon.txt]: Done

File: src/test/resources/templates/library_csv_template_output.txt (hunk 7 above)
Comment (ghost) [reply src/test/resources/templates/library_csv_template_output.txt]: Fixed

File: src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java (hunk 3 above)
Comment (ghost) [reply src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java]: Done

File: src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java (hunk 8 above)
Comment (ghost) [reply src/integrationTest/java/com/epam/fonda/RnaExpressionFastqIntegrationTest.java]: Done

File: src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java [hunk 9]
```diff
@@ -90,20 +92,37 @@ public void cleanOutputDirForNextTest(String outputDir) throws IOException {
...
+                    System.lineSeparator().equalsIgnoreCase(CRLF.getLineSeparator()) ? CRLF : LF;
             final String source = new String(Files.readAllBytes(Paths.get(globalConfig))) +
                     String.format("%nline_ending = %s", lineSeparator.name());
             writeToFile(globalConfig, source, lineSeparator);
+
+            if (ArrayUtils.isNotEmpty(nonRequiredOptions) && Arrays.asList(nonRequiredOptions).contains("-master")) {
+                context.setVariable("master", true);
+            } else {
+                context.setVariable("master", false);
+            }
```
Comment (kamyshova) [inline src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java]: You can extract 'master' mode check to separate method and simplify this to only `context.setVariable("master", checkOption("-master"));`

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java (hunk 9 above)
Comment (ghost) [reply src/integrationTest/java/com/epam/fonda/AbstractIntegrationTest.java]: Done

**Review by kamyshova — APPROVED** [review]

**Pull request merged**